resources/definitions/weedo_x40.def.json

```diff
@@ -0,0 +1,334 @@
+        "variants_name": "Nozzle Type",
+        "machine_extruder_trains":
+        {
+            "0": "weedo_x40_extruder_left_0",
+            "1": "weedo_x40_extruder_right_0"
+        },
+        "supports_usb_connection": true
+    },
+    
+    "overrides": {
+        "machine_name": { "default_value": "Weedo X40" },
+        "machine_width": { "default_value": 300 },
+        "machine_depth": { "default_value": 300 },
+        "machine_height": { "default_value": 400 },
+        "machine_heated_bed": { "default_value": true },
+        "machine_nozzle_heat_up_speed": { "default_value": 6 },
+        "machine_nozzle_cool_down_speed": { "default_value": 4 },
+        "machine_head_with_fans_polygon":
+        {
+            "default_value":
+            [
+                [ -42.5, 18 ],
+                [ -42.5, -50 ],
+                [ 24.5, 18 ],
+                [ 24.5, -50.0 ]
+            ]
+        },
+        "machine_max_feedrate_x": { "value": 500 },
+        "machine_max_feedrate_y": { "value": 500 },
+        "machine_max_feedrate_z": { "value": 10 },
+        "machine_max_feedrate_e": { "value": 50 },       
+        "machine_acceleration": { "value": 500 },
+        "machine_max_acceleration_x": { "value": 500 },
+        "machine_max_acceleration_y": { "value": 500 },
+        "machine_max_acceleration_z": { "value": 100 },
+        "machine_max_acceleration_e": { "value": 5000 },
+        "gantry_height": { "value": "12" },
+        "machine_extruder_count": { "default_value": 2 },
+        "machine_start_gcode": {"default_value": "; x40-community.org configuration Rev. 07\n;(**** start.gcode for WEEDO X40 DUAL****)\nT{initial_extruder_nr} S ; Selected start extruder\nM140 S{material_bed_temperature_layer_0} ; Preheat bed\nM109 S{material_print_temperature_layer_0}; Preheat nozzle\nM73 P0 ; Set current print progress percentage\nG21 ; Millimeter Units\nG90 ; Absolute positioning\nM82 ; Extruder in absolute mode\nT0 S ; Select left extruder\nM301 H1 P15.53 I1.32 D45.75 ; PID left extruder with Weedo X40 coolingsystem\n;M301 H1 P13.32 I0.98 D45.13 ; PID left extruder with X40 Community coolingsystem\nM92 E94.90 ; Calibrate left extruder\nT1 S ; Select right extruder\nM301 H1 P15.44 I1.29 D46.11 ; PID right extruder with Weedo X40 coolingsystem\n;M301 H1 P13.32 I0.98 D45.13 ; PID right extruder with X40 Community coolingsystem\nM92 E94.90 ; Calibrate right extruder\nT0 S ; Select left extruder\nG28 ; Auto home\nM420 S ; Actived bed leveling\nG29 ; Bed Leveling\nG1 X-47 F3000 ; Move left nozzle to parking position\nT1 S ; select right extruder\nG1 X352 F3000 ; Move right nozzle to parking position\nM107 ; Turn off fan\nT{initial_extruder_nr} S ; Set start extruder\nM190 S{material_bed_temperature_layer_0} ; Waiting for bed temperature\nM192 S{material_print_temperature_layer_0} ; Waiting for nozzle temperature\nG1 E50 F100 ; Extrude in parking position\nM77 ; Stop heat up timer\nM75 ; Start print timer\n" 
+        },

```

Comment:
You included M301 PID values for your custom cooling system.
M420 not needed, G29 will automatically enable bed leveling
M107 need P0/P1 or it'll only control the left fan
Typrically M109 is used for hotend temperature, not M192